Core/src/IfcGeometryConverter/ProfileConverter.h

```diff
@@ -69,47 +69,49 @@ namespace OpenInfraPlatform {
 					convertIfcArbitraryClosedProfileDef(arbitrary_closed, paths);
 					removeDuplicates(paths);
 					return;
 				}
 
-				// (2/5) IfcArbitraryOpenProfileDef SUBTYPE OF IfcProfileDef
-				//std::shared_ptr<typename IfcEntityTypesT::IfcArbitraryOpenProfileDef> arbitrary_open =
-				//	std::dynamic_pointer_cast<typename IfcEntityTypesT::IfcArbitraryOpenProfileDef>(profileDef);
+				// (2/8) IfcArbitraryOpenProfileDef SUBTYPE OF IfcProfileDef
 				if (profileDef.isOfType<typename IfcEntityTypesT::IfcArbitraryOpenProfileDef>()) {
 					EXPRESSReference<typename IfcEntityTypesT::IfcArbitraryOpenProfileDef> arbitrary_open = profileDef.as<typename IfcEntityTypesT::IfcArbitraryOpenProfileDef>();
 					convertIfcArbitraryOpenProfileDef(arbitrary_open, paths);
 					removeDuplicates(paths);
 					return;
 				}
 
-				// (3/5) IfcCompositeProfileDef SUBTYPE OF IfcProfileDef
-				//std::shared_ptr<typename IfcEntityTypesT::IfcCompositeProfileDef> composite = std::dynamic_pointer_cast<typename IfcEntityTypesT::IfcCompositeProfileDef>(profileDef);
+				// (3/8) IfcCompositeProfileDef SUBTYPE OF IfcProfileDef
 				if (profileDef.isOfType<typename IfcEntityTypesT::IfcCompositeProfileDef>()) {
 					EXPRESSReference<typename IfcEntityTypesT::IfcCompositeProfileDef> composite = profileDef.as<typename IfcEntityTypesT::IfcCompositeProfileDef>();
 					convertIfcCompositeProfileDef(composite, paths);
 					removeDuplicates(paths);
 					return;
 				}
 
-				// (4/5) IfcDerivedProfileDef SUBTYPE OF IfcProfileDef
-				//std::shared_ptr<typename IfcEntityTypesT::IfcDerivedProfileDef> derived = std::dynamic_pointer_cast<typename IfcEntityTypesT::IfcDerivedProfileDef>(profileDef);
+				// (4/8) IfcDerivedProfileDef SUBTYPE OF IfcProfileDef
 				if (profileDef.isOfType<typename IfcEntityTypesT::IfcDerivedProfileDef>()) {
 					EXPRESSReference<typename IfcEntityTypesT::IfcDerivedProfileDef> derived = profileDef.as<typename IfcEntityTypesT::IfcDerivedProfileDef>();
 					convertIfcDerivedProfileDef(derived, paths);
 					removeDuplicates(paths);
 					return;
 				}
 
-				// (5/5) IfcParameterizedProfileDef SUBTYPE OF IfcProfileDef
-				//std::shared_ptr<typename IfcEntityTypesT::IfcParameterizedProfileDef> parameterized =
-				//	std::dynamic_pointer_cast<typename IfcEntityTypesT::IfcParameterizedProfileDef>(profileDef);
+				// (5/8) IfcOpenCrossProfileDef SUBTYPE OF IfcProfileDef (exists starting IFC4x3)
+#if defined(OIP_MODULE_EARLYBINDING_IFC4X1) || defined(OIP_MODULE_EARLYBINDING_IFC4) || defined(OIP_MODULE_EARLYBINDING_IFC2X3)
+#else
+				if (profileDef.isOfType<typename IfcEntityTypesT::IfcOpenCrossProfileDef>()) {
+					EXPRESSReference<typename IfcEntityTypesT::IfcOpenCrossProfileDef> open_cross = profileDef.as<typename IfcEntityTypesT::IfcOpenCrossProfileDef>();
+					convertIfcOpenCrossProfileDef(open_cross, paths);
+					removeDuplicates(paths);
+					return;
+				}
+#endif
+				// (6/8) IfcParameterizedProfileDef SUBTYPE OF IfcProfileDef

```

Comment:
How come it's 6/8? Where are the other two?

Reply:
Resolved: afdd101